### PR TITLE
[jsgraph] new jsgraph type

### PR DESCRIPTION
**Pull request by rosinbum**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/jsgraph/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        visitor: Visitor,
+        options?: TraversalOptions
+    }
+
+    export interface TraversalResult {
+        error?: any;
+        result?: TraversalContext
+    }
+
+    export type colors = { white: 0, gray: 1, black: 2 };
```
Comment (DanielRosenwasser) [inline types/jsgraph/index.d.ts]: Should `colors` be capitalized?

File: types/jsgraph/tslint.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "adjacent-overload-signatures": false,
```
Comment (DanielRosenwasser) [inline types/jsgraph/tslint.json]: Can you re-enable these rules?

File: types/jsgraph/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
+// Type definitions for jsgraph 0.7
+// Project: https://github.com/Encapsule/jsgraph
+// Definitions by: Joel Rosinbum <https://github.com/rosinbum>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+
```
Comment (DanielRosenwasser) [inline types/jsgraph/index.d.ts]: Nit: Can you remove the double newlines?

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/jsgraph/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        hasEdgeProperty(edge: Edge): boolean;
+
+        clearEdgeProperty(edge: Edge): boolean;
+
+
+    }
+
+    export interface Visitor {
+        initializeVertex(req: { u: string, g: DirectedGraph }): void;
```
Comment (DanielRosenwasser) [inline types/jsgraph/index.d.ts]: Seems like you could extract `{ u: string, g: DirectedGraph }` to its own type.

File: types/jsgraph/index.d.ts (hunk 1 above)
Comment (rosinbum) [reply types/jsgraph/index.d.ts]: nope :)

File: types/jsgraph/tslint.json (hunk 2 above)
Comment (rosinbum) [reply types/jsgraph/tslint.json]: sure

File: types/jsgraph/index.d.ts (hunk 4 above)
Comment (rosinbum) [reply types/jsgraph/index.d.ts]: will do

**Comment by DanielRosenwasser** [comment]
I'd rather not hold this up, so I'm going to merge this and you can send the updates if you'd like.

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**